### PR TITLE
Initialize ObjectValue with null

### PR DESCRIPTION
**Pull request by jderusse**
Alternative to #1079

### CODE REVIEW

File: src/CodeGenerator/src/Generator/ResponseParser/RestJsonParser.php [hunk 1]
```diff
@@ -112,15 +112,15 @@ private function getInputAccessorName(Member $member)
         return $member->getLocationName();
     }
 
-    private function parseElement(string $input, Shape $shape, bool $required)
+    private function parseElement(string $input, Shape $shape, bool $required, bool $inObject = false)
```
Comment (Nyholm) [inline src/CodeGenerator/src/Generator/ResponseParser/RestJsonParser.php:115]: Let's make this non optional. Also, can we add a PHP doc block on this method to describe `$inObject`?

File: src/CodeGenerator/src/Generator/ResponseParser/RestJsonParser.php [hunk 2]
```diff
@@ -218,7 +218,7 @@ private function parseResponseBlob(string $input, bool $required): string
         return strtr('isset(INPUT) ? base64_decode((string) INPUT) : null', ['INPUT' => $input]);
     }
 
-    private function parseResponseList(ListShape $shape, string $input, bool $required): string
+    private function parseResponseList(ListShape $shape, string $input, bool $required, bool $inObject = false): string
```
Comment (Nyholm) [inline src/CodeGenerator/src/Generator/ResponseParser/RestJsonParser.php:221]: Same here. This is only used once, no default needed.

File: src/CodeGenerator/src/Generator/ResponseParser/RestJsonParser.php [hunk 3]
```diff
@@ -250,18 +250,20 @@ private function parseResponseList(ListShape $shape, string $input, bool $requir
...
+        return strtr($required ? '$this->FUNCTION_NAME(INPUT)' : 'EMPTY_METHOD(INPUT) ? EMPTY : $this->FUNCTION_NAME(INPUT)', [
+            'EMPTY_METHOD' => $inObject ? '!isset' : 'empty',
+            'EMPTY' => !$inObject ? '[]' : 'null',
             'INPUT' => $input,
             'FUNCTION_NAME' => $functionName,
         ]);
     }
 
-    private function parseResponseMap(MapShape $shape, string $input, bool $required): string
+    private function parseResponseMap(MapShape $shape, string $input, bool $required, bool $inObject = false): string
```
Comment (Nyholm) [inline src/CodeGenerator/src/Generator/ResponseParser/RestJsonParser.php:266]: This is also only used once.

File: src/CodeGenerator/src/Generator/ResponseParser/RestXmlParser.php [hunk 4]
```diff
@@ -125,15 +125,15 @@ private function getInputAccessor(string $currentInput, Member $member)
         return $currentInput . ($member->getLocationName() ? '->' . $member->getLocationName() : '');
     }
 
-    private function parseXmlElement(string $input, Shape $shape, bool $required)
+    private function parseXmlElement(string $input, Shape $shape, bool $required, bool $inObject = false)
```
Comment (Nyholm) [inline src/CodeGenerator/src/Generator/ResponseParser/RestXmlParser.php:128]: Same.

**Review by Nyholm — APPROVED** [review]
I like this one better. 
Just a small fix

File: src/CodeGenerator/src/Generator/ResponseParser/RestJsonParser.php (hunk 1 above)
Comment (jderusse) [reply src/CodeGenerator/src/Generator/ResponseParser/RestJsonParser.php]: fixed

**Review by Nyholm — APPROVED** [review]
Cool. Thank you

**Pull request merged**